src/js/types/networks/ethereum.js

```diff
@@ -101,22 +101,34 @@ type EthereumSignTypedDataMessage<T: EthereumSignTypedDataTypes> = {
     message: { [fieldName: string]: any },
 };
 
+/**
+ * Used for full EIP-712 signing
+ * (currently only supported on Trezor Model T)
+ */
 export type EthereumSignTypedData = {
     path: string | number[],
     metamask_v4_compat: boolean,
-    data: EthereumSignTypedDataMessage<any>,
-    domain_separator_hash?: typeof undefined,
-    message_hash?: typeof undefined,
+    data: EthereumSignTypedDataMessage<EthereumSignTypedDataTypes>,
 };
 
+/**
+ * Used for EIP-712 blind signing on Trezor Model 1 only
+ */
 export type EthereumSignTypedHash = {
     path: string | number[],
-    metamask_v4_compat?: typeof undefined,
-    data?: typeof undefined,
     domain_separator_hash: string,
     message_hash: string,
 };
 
+/**
+ * Used for full EIP-712 signing or blind signing.
+ * Supports both Trezor Model T and Trezor Model 1
+ */
+export type EthereumSignTypedHashOrData = {
+    ...$Exact<EthereumSignTypedData>,
+    ...$Exact<EthereumSignTypedHash>,
+};
+
```

Comment:
can we maybe have this also in ethereum.d.ts?

Reply:
Done in https://github.com/trezor/connect/commit/7139d792a495d4db5cfd20ae63d1e1c3c03539a9

I've also renamed this type to `EthereumSignTypedHashAndData` instead of `...HashOrData`, to make it more clear that it should have both the typed hash AND the typed data.